cmd/abapEnvironmentRunAUnitTest.go

```diff
@@ -0,0 +1,569 @@
+	}
+	if AUnitConfig.Options.Scope.ForeignTests != nil {
+		optionsString += ` foreignTests="` + fmt.Sprintf("%v", *AUnitConfig.Options.Scope.ForeignTests) + `"`
+	}
+	optionsString += `/><aunit:riskLevel`
+	if AUnitConfig.Options.RiskLevel.Harmless != nil {
+		optionsString += ` harmless="` + fmt.Sprintf("%v", *AUnitConfig.Options.RiskLevel.Harmless) + `"`
+	}
+	if AUnitConfig.Options.RiskLevel.Dangerous != nil {
+		optionsString += ` dangerous="` + fmt.Sprintf("%v", *AUnitConfig.Options.RiskLevel.Dangerous) + `"`
+	}
+	if AUnitConfig.Options.RiskLevel.Critical != nil {
+		optionsString += ` critical="` + fmt.Sprintf("%v", *AUnitConfig.Options.RiskLevel.Critical) + `"`
+	}
+	optionsString += `/><aunit:duration`
+	if AUnitConfig.Options.Duration.Short != nil {
+		optionsString += ` short="` + fmt.Sprintf("%v", *AUnitConfig.Options.Duration.Short) + `"`
+	}
+	if AUnitConfig.Options.Duration.Medium != nil {
+		optionsString += ` medium="` + fmt.Sprintf("%v", *AUnitConfig.Options.Duration.Medium) + `"`
+	}
+	if AUnitConfig.Options.Duration.Long != nil {
+		optionsString += ` long="` + fmt.Sprintf("%v", *AUnitConfig.Options.Duration.Long) + `"`
+	}
+	optionsString += `/></aunit:options>`
+	return optionsString
+}
+
+func buildAUnitObjectSetString(AUnitConfig AUnitConfig) (objectSetString string) {
+
+	//Build ObjectSets {
+	for _, s := range AUnitConfig.ObjectSet {
+		//Write object set
+		objectSetString += `<osl:objectSet xsi:type="` + s.Type + `" xmlns:osl="http://www.sap.com/api/osl" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">`
+		for _, t := range s.Set {
+			objectSetString += `<osl:set xsi:type="` + t.Type + `">`
+			for _, packageSet := range t.PackageSet {
+				objectSetString += `<osl:package name="` + packageSet.Name + `" includeSubpackages="` + fmt.Sprintf("%v", *packageSet.IncludeSubpackages) + `"/>`
+			}
+			for _, flatObjectSet := range t.FlatObjectSet {
+				objectSetString += `<osl:package name="` + flatObjectSet.Name + `" includeSubpackages="` + fmt.Sprintf("%v", *&flatObjectSet.Type) + `"/>`
+			}
+			objectSetString += `</osl:set>`
+		}
+
+		objectSetString += `</osl:objectSet>`
+	}
+	return objectSetString
+}
+
+func fetchAUnitXcsrfToken(requestType string, details abaputils.ConnectionDetailsHTTP, body []byte, client piperhttp.Sender) (string, error) {
+
+	log.Entry().WithField("ABAP Endpoint: ", details.URL).Debug("Fetching Xcrsf-Token")
+
+	details.URL += "/sap/bc/adt/api/abapunit/runs/00000000000000000000000000000000"
+	details.XCsrfToken = "fetch"
+	header := make(map[string][]string)
+	header["X-Csrf-Token"] = []string{details.XCsrfToken}
+	header["Accept"] = []string{"application/vnd.sap.adt.api.abapunit.run-status.v1+xml"}
+	req, err := client.SendRequest(requestType, details.URL, bytes.NewBuffer(body), header, nil)
+	if err != nil {
+		return "", fmt.Errorf("Fetching Xcsrf-Token failed: %w", err)
+	}
+	defer req.Body.Close()
+
+	// workaround until golang version 1.16 is used
```

Comment:
As far as I know this is not necessary anymore. I built it back in most abapEnvironment steps